modules/tensor_mechanics/src/scalarkernels/GeneralizedPlaneStrainDiag.C

```diff
@@ -0,0 +1,46 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "GeneralizedPlaneStrainDiag.h"
+#include "Assembly.h"
+
+template<>
+InputParameters validParams<GeneralizedPlaneStrainDiag>()
+{
+  InputParameters params = validParams<ScalarKernel>();
+  params.addClassDescription("Generalized Plane Strain Scalar Kernel");
+  params.addRequiredParam<UserObjectName>("gps_uo", "The name of the GeneralizedPlaneStrainUO UserObject");
```

Comment:
I'm not a fan of this parameter name. maybe we can come up with something less cryptic and without a system suffix like `_uo`. Why not just `generalized_plane_strain`